server/broadcast.go

```diff
@@ -96,7 +96,12 @@ func processSegment(cxn *rtmpConnection, seg *stream.HLSSegment) {
 	if err != nil {
 		glog.Errorf("Error saving segment %d: %v", seg.SeqNo, err)
 		if monitor.Enabled {
-			monitor.LogSegmentUploadFailed(nonce, seg.SeqNo, monitor.SegmentUploadErrorUnknown, err.Error())
+			switch err.Error() {
```

Comment:
Since we're doing some specific error checking within `LogSegmentUploadFailed`, can we put this bit there too and streamline the flow at this spot?

Reply:
done